Update HDInsight domain-join instructions

syedhassaanahmed opened the pull request:
Current document is outdated due to the following reasons.

- Most `Azure AD` operations (apart from creating a new `AD`) can now be done without the need to navigate to classic portal.
- `Azure AD DS` now supports `Resource Manager` `VNets` so no need to create classic `VNets` and peer them with their `RM` counterparts.
- `Azure AD DS` `LDAPS` now supports self-signed certificates.
- HDInsight premium is now supported on `Spark` and `Interactive Query clusters` as well.

azurecla commented:
Hi __@v-anpasi__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

syedhassaanahmed committed:
Update HDInsight domain-join instructions

Current document is outdated due to the following reasons.
- Most Azure AD operations (apart from creating a new Active Directory) can now be done without the need to navigate to classic portal.
- Azure AD DS now supports Resource Manager VNets so no need to create classic VNets and peer them with their RM counterparts.
- Azure AD DS LDAPS now supports self-signed certificates.
- HDInsight premium is now supported on Spark and Interactive Query clusters as well.

syedhassaanahmed committed:
Remove outdated topology diagram

GitHubber17 committed:
Update apache-domain-joined-configure.md
